hummingbot/market/kucoin/kucoin_api_order_book_data_source.py

```diff
@@ -0,0 +1,282 @@
+        # Get the currently active markets
+        async with aiohttp.ClientSession() as client:
+            trading_pairs: List[str] = await self.get_trading_pairs()
+            retval: Dict[str, OrderBookTrackerEntry] = {}
+
+            number_of_pairs: int = len(trading_pairs)
+            for index, trading_pair in enumerate(trading_pairs):
+                try:
+                    snapshot: Dict[str, Any] = await self.get_snapshot(client, trading_pair)
+                    snapshot_msg: OrderBookMessage = KucoinOrderBook.snapshot_message_from_exchange(
+                        snapshot,
+                        metadata={"symbol": trading_pair}
+                    )
+                    order_book: OrderBook = self.order_book_create_function()
+                    order_book.apply_snapshot(snapshot_msg.bids, snapshot_msg.asks, snapshot_msg.update_id)
+                    retval[trading_pair] = OrderBookTrackerEntry(trading_pair, snapshot_msg.timestamp, order_book)
+                    self.logger().info(f"Initialized order book for {trading_pair}. "
+                                       f"{index + 1}/{number_of_pairs} completed.")
+                    # Kucoin rate limit is 100 https requests per 10 seconds
+                    await asyncio.sleep(0.4)
+                except Exception:
+                    self.logger().error(f"Error getting snapshot for {trading_pair}. ", exc_info=True)
+                    await asyncio.sleep(10)
+            return retval
+
+    async def _inner_messages(self,
+                              ws: websockets.WebSocketClientProtocol) -> AsyncIterable[str]:
+        # Terminate the recv() loop as soon as the next message timed out, so the outer loop can reconnect.
+        try:
+            while True:
+                try:
+                    msg: str = await asyncio.wait_for(ws.recv(), timeout=self.MESSAGE_TIMEOUT)
+                    yield msg
+                except asyncio.TimeoutError:
+                    try:
+                        pong_waiter = await ws.ping()
+                        await asyncio.wait_for(pong_waiter, timeout=self.PING_TIMEOUT)
+                    except asyncio.TimeoutError:
+                        raise
+        except asyncio.TimeoutError:
+            self.logger().warning("WebSocket ping timed out. Going to reconnect...")
+            return
+        except ConnectionClosed:
+            return
+        finally:
+            await ws.close()
+			
+	#get required data to create a websocket request
+	async def ws_connect_data(self):
+		async with aiohttp.ClientSession() as session:
+			async with session.post('https://api.kucoin.com/api/v1/bullet-public', data=b'') as resp:
+				response: aiohttp.ClientResponse = resp
+				if response.status != 200:
+					raise IOError(f"Error fetching Kucoin websocket connection data."
+								  f"HTTP status is {response.status}.")
+				api_data = await response.read()
+				data: Dict[str, Any] = json.loads(api_data)
```

Comment:
Small matter,
Considering that KuCoin documentation did mention that >All requests and responses are application/json content type.

You can simply do the following:
`data: Dict[str, Any] = await response.json()`